instrumentation/github.com/aws/aws-lambda-go/otellambda/config.go

```diff
@@ -92,6 +93,20 @@ type config struct {
 	Propagator propagation.TextMapPropagator
 }
 
+type asyncSafeFlusher struct {
+	flusher Flusher
+}
+
+func (f asyncSafeFlusher) ForceFlush(ctx context.Context) error {
+	// yield processor to attempt to ensure all spans have
+	// been consumed and are ready to be flushed
+	// - see https://github.com/open-telemetry/opentelemetry-go/issues/2080
+	// to be removed upon resolution of above issue
+	runtime.Gosched()
+
+	return f.flusher.ForceFlush(ctx)
+}
+
```

Comment:
I have just landed https://github.com/open-telemetry/opentelemetry-go/pull/2335 which will enable us to remove this entirely and simply call `ForceFlush` on the provided `TracerProvider`.  The entire `Flusher` type and its implementations and configuration options can be removed.